src/controllers/__tests__/ngosController.test.js

```diff
@@ -1,73 +1,128 @@
-    username: 'Some Fancy Username',
-    website: 'https://www.womefanctwebsite.org',
-    email: 'info@nmn.edu.tr(1)',
+    username: 'SomeFancyUsername2',
+    email: 'info@nmn.edu.tr',
     password: 'some passward',
-    phone: '0216123456789',
-    publishedEvents: ['objectId(2)', 'objectId(3)'],
-    avatar: 'https://www.womefncyavatar.png',
+    type: 'Ngo',
   },
 ];
+const events = [
+  {
+    avatar:
+      'https://www.estidia.eu/wp-content/uploads/2018/04/free-png-upcoming-events-clipart-icons-for-calendar-of-events-800.png',
+    description:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+    location: 'Antalya',
+    launchDate: new Date(),
+    ngo: '62e9008803b4427103cb4462',
+    topic: 'Coding',
+    pendingApplicants: [],
+  },
+  {
+    avatar:
+      'https://www.estidia.eu/wp-content/uploads/2018/04/free-png-upcoming-events-clipart-icons-for-calendar-of-events-800.png',
+    description:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+    location: 'İstanbul',
+    launchDate: new Date(),
+    ngo: '62e9008803b4427103cb4462',
+    topic: 'English',
+    pendingApplicants: [],
+  },
+];
+const ngoId = '62e9008803b4427103cb4462';
 
-beforeAll((done) => {
-  done();
+beforeEach(async () => {
+  await Event.deleteMany();
+  await Volunteer.deleteMany();
+  await Ngo.deleteMany();
 });
-
-afterAll((done) => {
-  done();
+afterEach(async () => {
+  await Event.deleteMany();
+  await Volunteer.deleteMany();
+  await Ngo.deleteMany();
+});
+afterAll(async () => {
+  await mongoose.connection.close();
 });
 
-describe("Testing ngos for routes doesn't require auth controls", () => {
-  it('GET /api/ngos should retrieve all the ngos', (done) => {
-    request(app)
-      .get(`/api/ngos/`)
+describe("Testing Ngos for routes doesn't require auth controls", () => {
+  it('GET /api/ngos/:id/events should retrieve all the events that are published by the specified NGO', async () => {
+    const ngo = await Ngo.create(mNgos[0]);
+
+    const mEvent1 = await Event.create(events[0]);
+    const mEvent2 = await Event.create(events[1]);
+
+    await Ngo.findOneAndUpdate(
+      { _id: mongoose.Types.ObjectId(ngo._id) },
```

Comment:
You don't need to wrap `ngo._id` with an ObjectId, since it's already an ObjectId.
```      { _id: mongoose.Types.ObjectId(ngo._id) },```